nixio/dimensions.py

```diff
@@ -251,14 +251,14 @@ def index_of(self, position):
         :rtype: int
         """
         ticks = self.ticks
-        if position < ticks[0]:
+        if position <= ticks[0]:
             return 0
-        elif position > ticks[-1]:
+        elif position >= ticks[-1]:
             return len(ticks) - 1
 
         ticks = np.array(ticks)
-        pidxs = np.flatnonzero((ticks - position) <= 0)
-        return int(pidxs[-1])
+        pidxs = np.searchsorted(ticks, position, side="right") - 1
```

Comment:
As in general searchsorted function will return a index that is 1 higher than what we want, so we have to do the -1 thing. But by default (which is left), there is one exception - namely when the position is exactly stepping on one of the ticks. if "right" is specified, the function returns i in ticks[i-1]<pos<=ticks[i] instead of ticks[i-1]<=pos<ticks[i]. This keep the returned value correct.